boards/openhwgroup/cv64a6_genesys_2/doc/index.rst

```diff
@@ -0,0 +1,80 @@
+.. zephyr:board:: cv64a6_genesys_2
+
+Overview
+********
+
+The Digilent Genesys 2 board features a Xilinx Kintex-7 FPGA which can run various softcore CPUs.
+In this configuration, the Genesys 2 is configured with a 64-bit version of the CVA6 RISC-V CPU.
+The SoC is configured with a memory controller interfacing with the Genesys' DRAM, PLIC and CLINT
+interrupt controllers, a UART device interfacing with the Genesys' USB UART, a RISC-V compatible
+debug module that interfaces with the Genesys' FTDI (USB JTAG) chip, a Xilinx SPI interface
+interfacing with the Genesys' SD card slot and a Xilinx GPIO interfacing with the Genesys' LEDs
+and switches.
+The complete hardware sources (see first reference) in conjunction with
+instructions for compiling and loading the configuration onto the Genesys 2 are available.
+
+See the following references for more information:
+
+- `CVA6 documentation`_
+- `Genesys 2 Reference Manual`_
+- `Genesys 2 Schematic`_
+
+Hardware
+********
+
+- CVA6 CPU with RV64imafdc instruction sets and an SV39 MMU
+- 1 GB DDR3 DRAM
+- 10/100/1000 Ethernet with copper interface, lowRISC Ethernet MAC
+- ns16550a-compatible USB UART, 115200 baud
+- RISCV debug module, connected via on-board FTDI (USB JTAG)
+- Xilinx SPI controller, connected to microSD slot
+- Xilinx GPIO, connected to 7 switches and LEDs
+
+Supported Features
+==================
+
+.. zephyr:board-supported-hw::
+
+Programming and Debugging
+*************************
+
+Loading the FPGA configuration
+==============================
+
+You need to build a bitstream with Xilinx Vivado and load it into the FPGA
+before you can load zephyr onto the board.
+Please refer to the CVA6 documentation for the required steps.
+This configuration is compatible with the following build target: cv64a6_imafdc_sv39
+
+Flashing
+========
+west flash is supported via the openocd runner.
+Here is an example for the :zephyr:code-sample:`hello_world` application.
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: cv64a6_genesys_2
+   :goals: build flash
```

Comment:
@WorldofJARcraft have you actually tested this? It doesn't seem like it's possible to build a hello world at the moment?
Also, you will want to add twister files so that this board is tested in CI, this will allow to catch such issues in the future. Somehow this was missed during the review of this PR.

I tried to kick things off here but you're probably better positioned to take it it from here :)

Reply:
@kartben I have tested the configuration, but against an outdated main branch, sorry.
Apparently, at some point during when I initially composed the PR and now, the device tree requirements for the RISC-V machine timer changed.
I have force-pushed a corrected version to the branch I have been using for this PR. I am also making the necessary changes for integrating the board into twister.
Should I make a new PR with the necessary changes or can I somehow overwrite the commits that have already been merged?

Reply:
@WorldofJARcraft Thanks for the quick feedback, really appreciated! It needs to be a new branch/PR based off of current main. FWIW what you now have in your branch looks in a much better shape but I encourage you to actually *run* Twister to check where things are at (for example I don't think you want "simulation: qemu" in there)

```
west twister -p cv32a6_genesys_2
```

Reply:
@kartben I have made a new PR for this: https://github.com/zephyrproject-rtos/zephyr/pull/88511
I have successfully tested the new PR in hardware against cv64a6.
I will set up and test on cv32a6 next week.
Until then, I will leave the PR in draft mode.

Reply:
Very nice, thanks for the update!